### PR TITLE
internals: Talk about ignition.platform.id

### PR DESCRIPTION
**Pull request by cgwalters**
I want to link to this from https://github.com/cgwalters/coreos-diskimage-rehydrator

### CODE REVIEW

**Review by jlebon — APPROVED** [review]
Nice!

File: internals/README-internals.md [hunk 1]
```diff
@@ -40,3 +40,31 @@ See also DHCP propagation: https://github.com/coreos/fedora-coreos-config/pull/4
...
+- [network requirement detection](https://github.com/coreos/fedora-coreos-config/blob/d87b52bc6a90b53e1afeab2731b52612d5e3bbc0/overlay.d/05core/usr/lib/dracut/modules.d/35coreos-network/coreos-enable-network.service#L13)
+
+Notice in particular how the time synchronization code ends up reconfiguring chrony dynamically.
+For other operating systems which do "per cloud" disk images, it would have been more
+natural to just change `/etc/chrony.conf` per platform.  But that would mean we have a different
+ostree commit checksum per platform, breaking our "image based" update model.
+
+It's very unlikely that we will change the platform IDs in the future.  However, FCOS users are recommended
+to avoid parsing `ignition.platform.id`.  Generally, higher level code that needs to be
+platform aware will have more platform-specific ways to find this information.
```
Comment (bgilbert) [inline internals/README-internals.md:69]: This seems a bit strong.  It's legitimate to have `ConditionKernelCommandLine=ignition.platform.id=x` in user-provided units.

Comment (cgwalters) [reply internals/README-internals.md:69]: I hesitated on this for sure but it seemed easier to say no than yes :smile:  If we're allowing it we should document that...somewhere?

Comment (lucab) [reply internals/README-internals.md:69]: https://docs.fedoraproject.org/en-US/fedora-coreos/platforms/

**Pull request merged**